haskell/experimental/private/module.bzl

```diff
@@ -347,10 +351,14 @@ def _build_haskell_module(
         progress_message = "HaskellBuildObject {} {}".format(hs.label, module.label),
         env = hs.env,
         arguments = args,
-        extra_name = module.label.package.replace("/", "_") + "_" + module.label.name,
```

Comment:
I'd leave this line as it was, unless you want to use the factored out `module_name` elsewhere.